### DISCUSSION TITLE
feat(datasource/custom): Support defaultRegistryUrlTemplate with local file with repository config ignored

### DISCUSSION THREAD

**Discussion by daveconde** [renovatebot/renovate#26010] [Suggest an Idea]
### Tell us more.

Currently a custom datasource that uses a local file needs to have the file located in the repository that renovate is running against due to the file being read using readLocalFile, it would be nice to be able to specify a file that can be used outside of the repository so that it is compatible with `requireConfig: "ignored"` and configuration specified in the global config.js passed to renovate.

**Comment by rarkins** [comment]
Allowing end users to read any file out of the entire file system is a security risk and I would prefer not to support it unless we find that this use case is more widely applicable to more users

**Reply by daveconde** [reply]
Maybe there's another way we could lock this down so its not so open? My usecase is specifically to have it passed via the config.js so I can generate the file when calling renovate via the cli.

**Reply by rarkins** [reply]
We always have to balance config bloat and complexity versus the benefit. If you are proposing something which (a) increases the security attack surface, and (b) increases the complexity of config options, and (c) is only for your use case, then unfortunately we need to decline it.

**Reply by daveconde** [reply]
Another approach could be to not load the file directly via defaultRegistryUrlTemplate but instead allow the data to be passed as a string property with the responsibility for loading outside of renovate. This would allow for dynamic generation of the data in a similar manner to how dynamic repositories are implemented here: https://github.com/renovatebot/renovate/issues/10011#issuecomment-1595418818. Would something like this address the security concerns?

**Reply by rarkins** [reply]
Potentially, but then you also need to make sure it doesn't add code bloat/complexity instead. I want to avoid adding complexity if it satisfies an extremely narrow use case.

**Reply by daveconde** [reply]
OK Thanks, I'll try and put a simple PR together over the next few days as a suggestion and we can see from there.

**Reply by daveconde** [reply]
For now I've implemented a small http service that I can run in the background as service in a GitLab pipeline, this enables me to leverage defaultRegistryUrlTemplate and inject the package versions I need via the shared gitlab /builds directory. Closing this off as it is enough for my use case at the moment.
